test/parallel_api/numeric/numeric.ops/exclusive_scan_by_segment.pass.cpp

```diff
@@ -85,13 +85,13 @@ void test_with_buffers()
 
 void test_with_usm()
 {
-    cl::sycl::queue q;
+    sycl::queue q;
     const int n = 10;
 
     // Allocate space for data using USM.
-    uint64_t* key_head = static_cast<uint64_t*>(cl::sycl::malloc_shared(n * sizeof(uint64_t), q.get_device(), q.get_context()));
-    uint64_t* val_head = static_cast<uint64_t*>(cl::sycl::malloc_shared(n * sizeof(uint64_t), q.get_device(), q.get_context()));
-    uint64_t* res_head = static_cast<uint64_t*>(cl::sycl::malloc_shared(n * sizeof(uint64_t), q.get_device(), q.get_context()));
+    uint64_t* key_head = static_cast<uint64_t*>(sycl::malloc_shared(n * sizeof(uint64_t), q.get_device(), q.get_context()));
```

Comment:
Have you considered using smart pointers? If we had a habit of using them, then we would not have faced deallocation issues.

```c++
auto sycl_deleter = [q](uint64_t* mem) { sycl::free(mem, q.get_context()); };
::std::unique_ptr<uint64_t[], decltype(sycl_deleter)> key_head(static_cast<uint64_t*>(sycl::malloc_shared(n * sizeof(uint64_t), q.get_device(), q.get_context())), sycl_deleter);
...
oneapi::dpl::exclusive_scan_by_segment(new_policy, key_head.get(), key_head.get()+n, val_head, res_head,
```
This is more difficult when just deallocating at the end of scope, but we can prevent such issue in the future if create a culture of using smart pointers.

Reply:
It could also be done with `std::vector<usm::alloc>` for example...

I'd suggest to start moving into the direction where we use `invoke_on_all_hetero_policies` and approach from `test1buffer`, `test2buffers`, `test3buffers` functions